File: GFG/JavaScript/DSA/level-order-traversal.js
// https://www.geeksforgeeks.org/problems/level-order-traversal/1

class Solution {
  //Function to return the level order traversal of a tree.
  levelOrder(root) {
    //your code here
    let res = [];
    let queue = [root];

    while (queue.length) {
      let node = queue.shift();
      res.push(node.data);

      if (node.left) {
        queue.push(node.left);
      }
      if (node.right) {
        queue.push(node.right);
      }
    }
    return res;
  }
}

File: GFG/Python/DSA/level-order-traversal.py
# https://www.geeksforgeeks.org/problems/level-order-traversal/1

class Solution:
    #Function to return the level order traversal of a tree.
    def level_order(self,root):
        # Code here
        res = []
        queue = [root]
        
        while len(queue):
            node = queue.pop(0)
            res.append(node.data)
            
            if node.left:
                queue.append(node.left)
            if node.right:
                queue.append(node.right)
        return res